pypesto/visualize/sampling.py

```diff
@@ -92,119 +112,347 @@ def sampling_fval_trace(
+                for condition_ids in all_condition_ids
+            ]).all()
+    ):
+        raise KeyError('All predictions must have the same set of conditions.')
+    condition_ids = all_condition_ids[0]
+
+    observable_ids = sorted({
+        observable_id
+        for prediction in ensemble_prediction.prediction_summary.values()
+        for condition in prediction.conditions
+        for observable_id in condition.observable_ids
+    })
+
+    # Set default labels.
+    labels = {
+        k: (labels[k] if k in labels else k)
+        for k in condition_ids + observable_ids
+    }
+
+    if groupby == CONDITION:
+        n_variables = len(observable_ids)
+        y_names = observable_ids
+        n_subplots = len(condition_ids)
+    elif groupby == OBSERVABLE:
+        n_variables = len(condition_ids)
+        y_names = condition_ids
+        n_subplots = len(observable_ids)
     else:
-        fig = ax.get_figure()
+        raise ValueError(f'Unsupported groupby value: {groupby}')
+
+    percentiles_opacity = sorted(
+        # min 30%, max 100%, opacity
+        np.linspace(0.3 * RGBA_MAX, RGBA_MAX, len(percentiles)),
+        reverse=True,
+    )
+    if axes is not None:
+        fig = axes.get_figure()
+        if not isinstance(axes, np.ndarray):
+            axes = np.array([[axes]])
+    cmap = plt.cm.viridis
+    cmap_min = RGBA_MIN
+    cmap_max = 0.85*(RGBA_MAX - RGBA_MIN) + RGBA_MIN  # exclude yellows
 
-    axes = dict(zip(ynames, ax.flat))
-
-    for i, level in enumerate(alpha_sorted):
-
-        # Get upper and lower bounds for the confidence level
-        lb, ub = calculate_prediction_profiles(values, alpha=level / 100)
-
-        # Get the median
-        _median = np.percentile(values, 50, axis=1)
-
-        n_timepoints = _median.shape[1]
-        n_conditions = _median.shape[0]
-
-        # Loop over observables/states
-        for j, iplot in enumerate(ynames):
-            # Create array for X axis
-            t = np.arange(n_timepoints)
-            # Loop over experimental conditions
-            for k in range(n_conditions):
-                # Plot confidence region
-                axes[iplot].fill_between(t,
-                                         ub[k, :, j],
-                                         lb[k, :, j],
-                                         facecolor=colors[i],
-                                         alpha=0.9,
-                                         label=str(level) + '% CI')
-                # Plot median
-                axes[iplot].plot(t, _median[k, :, j],
-                                 'k-', label='MCMC median')
-                axes[iplot].set_ylabel(yname + '_' + str(j + 1))
-                # Update X axis array
-                t += n_timepoints
+    # define colormap
+    variables_color = [
+        list(cmap(v))[:LEN_RGB]
+        for v in np.linspace(cmap_min, cmap_max, n_variables)
+    ]
+
+    if axes is None:
+        n_row = int(np.round(np.sqrt(n_subplots)))
```

Comment:
I think usually wider plots are preferable to higher plots. Thus, e.g. if n=6, here it would be 3 rows, 2 cols, right? Thus it may make sense to calculate the cols first?

Reply:
In the current state I think it would actually be 2 rows by 3 cols?